ultraviolet/shared/src/main/scala/ultraviolet/macros/CreateShaderAST.scala

```diff
@@ -645,6 +645,76 @@ class CreateShaderAST[Q <: Quotes](using val qq: Q) extends ShaderMacroUtils:
 
       // For loops
 
+      case Apply(
+            Apply(
+              TypeApply(Ident(forLoopName), _),
+              List(
+                Apply(
+                  TypeApply(Select(Ident(tupleIdent), "apply"), _),
+                  tupleArgs
+                ),
+                Block(
+                  List(DefDef("$anonfun", _, _, Some(condition))),
+                  Closure(Ident("$anonfun"), None)
+                ),
+                Block(
+                  List(DefDef("$anonfun", _, Inferred(), Some(Apply(_, nextArgs)))),
+                  Closure(Ident("$anonfun"), None)
+                )
+              )
+            ),
+            List(
+              Block(
+                Nil,
+                Block(
+                  List(
+                    DefDef(
+                      "$anonfun",
+                      List(TermParamClause(valdefs)),
+                      _,
+                      Some(body)
+                    )
+                  ),
+                  Closure(Ident("$anonfun"), None)
+                )
+              )
+            )
+          ) if (forLoopName == "cfor" || forLoopName == "_for") && isTuple.matches(tupleIdent) && valdefs.forall {
+            case ValDef(_, _, None) => true
+          } =>
+        val valNames = valdefs.map { case ValDef(name, _, _) =>
+          if name.contains("$") then proxies.makeVarName else name
+        }
+        val vals = tupleArgs.zip(valNames).map { case (init, name) =>
+          ShaderAST.Val(
+            name,
+            walkTerm(init, envVarName),
+            findReturnType(walkTerm(init, envVarName))
+          )
+        }
+        val retType   = findReturnType(vals.head)
+        val multiVals = ShaderAST.MultiStatements(retType, vals)
+
+        def replaceName: PartialFunction[ShaderAST, ShaderAST] = {
+          // TODO
+          case x =>
+            x
+        }
```

Comment:
> // TODO

The version from the existing for loop is:

```scala
        def replaceName: PartialFunction[ShaderAST, ShaderAST] = {
          case ShaderAST.DataTypes.ident(id) if id.startsWith("_") =>
            ShaderAST.DataTypes.ident(varName)
        }
```

Reply:
Just removed in https://github.com/PurpleKingdomGames/ultraviolet/pull/142/commits/8fd509bf9022a84112c2ea0d54b577997e86e1b7